### PR TITLE
Create GraphQL auth resolvers

### PR DESCRIPTION
**Pull request by sherryhli**
## Notion ticket link

[GraphQL auth resolvers](https://www.notion.so/uwblueprintexecs/GraphQL-auth-resolvers-082310793eb04e9598e82b81dd78dac7)

## Implementation description
* Mirrored auth REST endpoints implementation in #9 

## Steps to test
1. Run the application, then test using **Postman** (easier to view response headers than in the GraphQL playground). Simply send POST requests to `http://localhost:5000/graphql`, and set the body type to be GraphQL
image
2. Create a user if you don't have one, see instructions in #13 
3. Login by executing this mutation, check that you receive an access token and that a `Set-Cookie` header is returned with the refresh token.
```graphql
mutation {
  login(email: "your.email@domain.org", password: "your-password")
}
```
4. Refresh token by executing this mutation, ensure that a `Cookie: refreshToken=<insert-refreshToken>` header is sent (should have been auto-populated in Postman after previous mutation)
```graphql
mutation {
    refresh
}
```
5. Logout by executing this mutation
```graphql
mutation {
    logout(userId: "your-user-id")
}
```
6. Generate a password reset link by executing this mutation
```graphql
mutation {
    resetPassword(email: "your-user-id")
}
```

## What should reviewers focus on?
* backend/typescript/graphql/resolvers/authResolvers.ts

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/graphql/types/authType.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { gql } from "apollo-server-express";
+
+const authType = gql`
+  extend type Mutation {
+    login(email: String!, password: String!): String!
```
Comment (alexguo8) [inline backend/typescript/graphql/types/authType.ts:5]: Possible improvement, could return a Response type that contains an "ok" field, so on frontend we can check if mutation succeeded or not
Not sure if this applies though if there won't be any errors

Comment (sherryhli) [reply backend/typescript/graphql/types/authType.ts:5]: My thought was to have the client do an error check by seeing if the `errors` field is present in the response, not sure if this is the usual practice for GraphQL though so let me know!

Comment (alexguo8) [reply backend/typescript/graphql/types/authType.ts:5]: Oh ok, I didn't know that we had the errors field, that should be good then

**Review by alexguo8 — APPROVED** [review]
Looks good to me

**Pull request merged**